### PR TITLE
TLS documentation is missing

### PR DESCRIPTION
**Pull request by ihsandemir**
Added documentation for TLS feature.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -58,6 +60,15 @@ We use --recursive flag for our dependency on googletest framework.
 
 First create a build directory from the root of the project. In the build directory, run the following commands depending on your environment.
 
+For compiling with SSL support: 
+- You need have the openssl installed in your development environment. Furthermore, the openssl include directory should be added to include directories and the openssl library directory should be listed in the link directories. Furthermore you need to set the openssl libraries to link.
```
Comment (Serdaro) [inline README.md]: To compile with SSL support:
- You need to have the openssl installed in your development environment: (i) Add openssl `include` directory to include directories, (ii) Add openssl `library` directory to the link directories list, (iii) Set the openssl libraries to link.

File: README.md [hunk 2]
```diff
@@ -58,6 +60,15 @@ We use --recursive flag for our dependency on googletest framework.
...
+ 
+- You can provide your openssl installation directory to cmake using the following flags: -DHZ_OPENSSL_INCLUDE_DIR="Path to open installation include directory" and -DHZ_OPENSSL_LIB_DIR="Path to openssl lib directory"
+
+- Use -DHZ_COMPILE_WITH_SSL=ON
+
+- Check top level CMakeLists.txt file to see which libraries we link for openssl in which environment. For Mac OS and Linux, we link with "ssl" and "crypto", for windows and example linkage for 64 bit library and release build: "libeay32MD ssleay32MD libcrypto64MD libSSL64MD".
+
```
Comment (Serdaro) [inline README.md]: Check the top level `CMakeLists.txt` file to see which libraries we link for openssl in which environment. For Mac OS and Linux, we link with "ssl" and "crypto", for Windows there is an example linkage for 64 bit library and release build: "libeay32MD ssleay32MD libcrypto64MD libSSL64MD".

File: README.md [hunk 3]
```diff
@@ -397,6 +408,114 @@ Reliable Topic implementation depends on the Ringbuffer data structure. The data
...
+```
+boost::shared_ptr<config::NearCacheConfig<int, std::string> > nearCacheConfig(
+            new config::NearCacheConfig<int, std::string>("myMap"));
+clientConfig..addNearCacheConfig(nearCacheConfig);
+```
+
+As for all configuration options, this near cache config should be performed before the client instance is obtained and the client config should not be altered after the client is instantiated.
+
+The NearCacheConfig class has a number of options including the EvictionConfig which determines the eviction strategy used. The options work exactly the same way as the Java client options. More information on the options can be found at http://docs.hazelcast.org/docs/3.7/manual/html-single/index.html#creating-near-cache-for-map and http://docs.hazelcast.org/docs/3.7/manual/html-single/index.html#map-eviction .
+
```
Comment (Serdaro) [inline README.md]: .......at http://docs.hazelcast.org/docs/latest/manual/html-single/index.html#creating-near-cache-for-map and http://docs.hazelcast.org/docs/latest/manual/html-single/index.html#map-eviction.

File: README.md [hunk 4]
```diff
@@ -397,6 +408,114 @@ Reliable Topic implementation depends on the Ringbuffer data structure. The data
...
+Examples are also provided for some options at the near cache folder under examples (also can be see at https://github.com/hazelcast/hazelcast-cpp-client/tree/master/examples).
+
+# TLS Feature
+Note: This feature only works when used with Enterprise Hazelcast server.
+
```
Comment (Serdaro) [inline README.md]: # TLS Feature

Note: This is a Hazelcast IMDG Enterprise feature.

File: README.md [hunk 5]
```diff
@@ -397,6 +408,114 @@ Reliable Topic implementation depends on the Ringbuffer data structure. The data
...
+You can encrypt all the communication between the client and the cluster using this feature. 
+This feature requires that you have installed openssl development library in your development environment.
+
```
Comment (Serdaro) [inline README.md]: It requires the openssl development library installed in your development environment.

File: README.md [hunk 6]
```diff
@@ -397,6 +408,114 @@ Reliable Topic implementation depends on the Ringbuffer data structure. The data
...
+You need to enable the SSL config in client network config. Furthermore, you should specify a correct path to the CA verification file for the trusted server. This path can be relative to the executable working directory. You can set the protocol type. The default protocol is TLSv1.2. The ssl config also lets you to set the cipher suite to be used.
+
```
Comment (Serdaro) [inline README.md]: ...The SSL config also lets you set the cipher suite to be used.

File: README.md (hunk 1 above)
Comment (ihsandemir) [inline README.md]: done.

File: README.md [hunk 7]
```diff
@@ -58,6 +60,15 @@ We use --recursive flag for our dependency on googletest framework.
 
 First create a build directory from the root of the project. In the build directory, run the following commands depending on your environment.
 
+For compiling with SSL support: 
+- You need have the openssl installed in your development environment. Furthermore, the openssl include directory should be added to include directories and the openssl library directory should be listed in the link directories. Furthermore you need to set the openssl libraries to link.
+ 
+- You can provide your openssl installation directory to cmake using the following flags: -DHZ_OPENSSL_INCLUDE_DIR="Path to open installation include directory" and -DHZ_OPENSSL_LIB_DIR="Path to openssl lib directory"
+
```
Comment (ihsandemir) [inline README.md]: done.

File: README.md (hunk 2 above)
Comment (ihsandemir) [inline README.md]: done.

File: README.md (hunk 3 above)
Comment (ihsandemir) [inline README.md]: done

File: README.md (hunk 4 above)
Comment (ihsandemir) [inline README.md]: ok

File: README.md (hunk 5 above)
Comment (ihsandemir) [inline README.md]: ok

File: README.md (hunk 6 above)
Comment (ihsandemir) [inline README.md]: ok

**Comment by Serdaro** [comment]
+1

File: README.md [hunk 8]
```diff
@@ -397,6 +415,112 @@ Reliable Topic implementation depends on the Ringbuffer data structure. The data
...
+Map can be configured to work with near cache enabled. Near cache allows local caching of entries fetched from the cluster at the client side. The local cache is updated when the client does a get request for a map entry. The locally cached entries are updated automatically as any change occurs in the map data in the cluster. Special internal event listeners are utilized for this purpose.
+
+If you are doing mostly reads from the map rather than map updates, then enabling near cache allows you to get the entries faster since it will return the locally cached entry which will eliminate any network interaction. But if your application is doing mostly writes (update, remove, etc.) from map, then you may not gain much by enabling the near cache since the updates will cause network traffic. 
+
+You can enable the near cache for a map by adding a near cache configuration for its name. For example:
+
+```
+boost::shared_ptr<config::NearCacheConfig<int, std::string> > nearCacheConfig(
+            new config::NearCacheConfig<int, std::string>("myMap"));
+clientConfig..addNearCacheConfig(nearCacheConfig);
```
Comment (sancar) [inline README.md]: Typo with two dots `..`

**Pull request merged**